config/aso/credentials.yaml

```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: aso-controller-settings
+type: Opaque
+data:
+  AZURE_SUBSCRIPTION_ID: ${AZURE_SUBSCRIPTION_ID_B64:=""}
+  AZURE_TENANT_ID: ${AZURE_TENANT_ID_B64:=""}
```

Comment:
Just curious, is there a reason we want to add the `:=""` part to the variable references?

Reply:
I admit this was a shameless copy-paste from here and didn't really consider whether it's necessary or not: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/fec6314ebb00355ecc617270f40c6367d7d4d75f/config/default/credentials.yaml#L7-L8

It looks like ASO will get stuck in a crash loop either way when the vars aren't defined, but with `:=""` it doesn't crash until an ASO resource is created, whereas without `:=""` it'll crash as soon as it starts up even when no ASO resources exist.

Reply:
For the CAPZ manager credentials, the idea is that you can optionally define global credentials but it's also possible to define per Cluster credentials via AzureClusterIdentity, so we don't want the credentials to be required when installing CAPZ as they aren't actually required until you create your first cluster (without the defaulting to "", which is what `:=""` does, clusterctl init would fail when the environment variables are not set).

Reply:
So I think with `:=""` is what we want then. Eventually, a set of ASO credentials will be generated for each individual cluster from an AzureClusterIdentity if one exists, so those users can still choose not to define the global credentials and ASO will work fine with CAPZ until the user tries to create an ASO resource themselves without specifying credentials and relying on the defaults.